src/lib.rs

```diff
@@ -102,31 +103,38 @@ impl UPClientZenoh {
     /// ```
     /// # async_std::task::block_on(async {
     ///     use up_client_zenoh::UPClientZenoh;
-    ///     use up_rust::{UEntity, UUri};
+    ///     use up_rust::{Number, UAuthority, UEntity, UUri};
     ///     use zenoh::config::Config;
-    ///     let uuri = UUri {
-    ///         entity: Some(UEntity {
-    ///             name: "default.entity".to_string(),
-    ///             id: Some(u32::from(rand::random::<u16>())),
-    ///             version_major: Some(1),
-    ///             version_minor: None,
-    ///             ..Default::default()
-    ///         })
-    ///         .into(),
+    ///     let uauthority = UAuthority {
+    ///         name: Some("MyAuthName".to_string()),
+    ///         number: Some(Number::Id(vec![1, 2, 3, 4])),
+    ///         ..Default::default()
+    ///     };
+    ///     let uentity = UEntity {
+    ///         name: "default.entity".to_string(),
+    ///         id: Some(u32::from(rand::random::<u16>())),
+    ///         version_major: Some(1),
+    ///         version_minor: None,
     ///         ..Default::default()
     ///     };
-    ///     let upclient = UPClientZenoh::new_with_uuri(Config::default(), uuri).await.unwrap();
+    ///     let upclient = UPClientZenoh::new_with_uuri(Config::default(), uauthority, uentity).await.unwrap();
     /// # });
     /// ```
     pub async fn new_with_uuri(
         config: Config,
-        source_uuri: UUri,
+        uauthority: UAuthority,
```

Comment:
I'm wondering if we should enforce that the `UAuthority` and `UEntity` passed in pass the Micro Form tests and return a failure otherwise? WDYT?

Reply:
Good catch! I added the validator and also use `testcase` to add some testcases, which I learned from up-rust :smile: